src/env/tock/storage.rs

```diff
@@ -330,44 +367,39 @@ impl UpgradeStorage for TockUpgradeStorage {
         if data.is_empty() {
             return Err(StorageError::OutOfBounds);
         }
-        let address = self.partition.start() + offset;
-        let write_range = ModRange::new(address, data.len());
-        if self.partition.contains_range(&write_range) {
+        if let Some(address) = self.partition.find_address(offset, data.len()) {
```

Comment:
Why not the following which avoids indentation?

```rust
let address = self.partition.find_address(offset, data.len()).ok_or(StorageError::OutOfBounds)?;
```

Reply:
Done